Update to use vaadin version 8.0.0.alpha4

ahie opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/dashboard-demo/10)
<!-- Reviewable:end -->

pleku commented:
Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[src/main/java/com/vaadin/demo/dashboard/component/ProfilePreferencesWindow.java, line 61 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrX_KsPnWJNIMIb0Zi:-KTrX_KsPnWJNIMIb0Zj:blo4r54) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/component/ProfilePreferencesWindow.java#L61)):*
> ```Java
>     private ComboBox<String> titleField;
>     @PropertyId("male")
>     private OptionGroup sexField;
> ```

This could be RadioButtonGroup now ?

---

*[src/main/java/com/vaadin/demo/dashboard/view/sales/SalesView.java, line 159 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTr_s6UbM1SNZdN04d3:-KTr_s6Vg9H59fgbV7da:b-dzjea0) ([raw file](https://github.com/vaadin/dashboard-demo/blob/e38a66d624bbc8d359042e6ce1990940d8adf8d4/src/main/java/com/vaadin/demo/dashboard/view/sales/SalesView.java#L159)):*
> ```Java
>         result.setZoomLevelsVisible(false);
>         result.setSizeFull();
>         result.setNoDataSourceCaption("<span class=\"v-label h2 light\">Add a data set from the dropdown above</span>");
> ```

What has happened to this caption?

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 49 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrzp1OPg7zkzozkD-O:-KTrzp1OPg7zkzozkD-P:bivws4g) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L49)):*
> ```Java
>     private static final DecimalFormat DECIMALFORMAT = new DecimalFormat(
>             "#.##");
>     private static final String[] DEFAULT_COLLAPSIBLE = { "country", "city",
> ```

These are never added to collapsibleColumns Set and it is always empty.

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 114 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrzUrIgE_VFqKvedH7:-KTrzUrIgE_VFqKvedH8:bvtacas) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L114)):*
> ```Java
>                         String filterValue = filter.getValue().trim()
>                                 .toLowerCase();
>                         return transaction.getCountry().toLowerCase()
> ```

What if any of these are null ? Used to be checked in the old filters

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 203 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-A7Yex3rWFpB-mod:-KTs-A7Yex3rWFpB-moe:bd1q1qk) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L203)):*
> ```Java
>         if (defaultColumnsVisible()) {
>             for (Column<Transaction, ?> column : collapsibleColumns) {
>                 if (Page.getCurrent().getBrowserWindowWidth() < 800) {
> ```

Can this change during the loop ? Why was this changed from the previous where it was much more simple

---

*[src/main/resources/com/vaadin/demo/dashboard/DashboardWidgetSet.gwt.xml, line 38 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-QMn6I7OswMd0EbG:-KTs-QMoVlcABrMZiUfE:byp18fd) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/resources/com/vaadin/demo/dashboard/DashboardWidgetSet.gwt.xml#L38)):*
> ```XML
>     
> 
>     <inherits name="com.vaadin.demo.dashboard.com.vaadin.demo.dashboard.DashboardWidgetSet" />
> ```

Remove

---

*[src/main/webapp/VAADIN/themes/dashboard/dashboard.scss, line 13 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-_9Q0V6mVj3F1oPd:-KTs-_9Q0V6mVj3F1oPe:b-s58lk1) ([raw file](https://github.com/vaadin/dashboard-demo/blob/e38a66d624bbc8d359042e6ce1990940d8adf8d4/src/main/webapp/VAADIN/themes/dashboard/dashboard.scss#L13)):*
> ```SCSS
> $v-included-components: remove($v-included-components, accordion);
> $v-included-components: remove($v-included-components, colorpicker);
> $v-included-components: remove($v-included-components, grid);
> ```

Can table be now removed ?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10)*
<!-- Sent from Reviewable.io -->

ahie commented:
Review status: 12 of 16 files reviewed at latest revision, 7 unresolved discussions.

---

*[src/main/java/com/vaadin/demo/dashboard/component/ProfilePreferencesWindow.java, line 61 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrX_KsPnWJNIMIb0Zi:-KTsLvOmFtDHV90nXemv:b-896fix) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/component/ProfilePreferencesWindow.java#L61)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> This could be RadioButtonGroup now ?
</details>
Done.

---

*[src/main/java/com/vaadin/demo/dashboard/view/sales/SalesView.java, line 159 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTr_s6UbM1SNZdN04d3:-KTsMABNNHnHU2axOOML:bhnpiqk) ([raw file](https://github.com/vaadin/dashboard-demo/blob/e38a66d624bbc8d359042e6ce1990940d8adf8d4/src/main/java/com/vaadin/demo/dashboard/view/sales/SalesView.java#L159)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> What has happened to this caption?
</details>
Is there a simple way to customise this in the new charts? As far as I can see this seems to be coupled with the Lang object and not worth the effort to configure.

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 49 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrzp1OPg7zkzozkD-O:-KTsFSbU9_RgFV4TAM2r:b-x6oq8a) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L49)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> These are never added to collapsibleColumns Set and it is always empty.
</details>
Added the correct columns to collapsibleColumns, removed the string array.

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 114 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTrzUrIgE_VFqKvedH7:-KTsWx_fpN7hyRuaBrQk:bqjfaei) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L114)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> What if any of these are null ? Used to be checked in the old filters
</details>
Improved to take nulls into account.

---

*[src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java, line 203 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-A7Yex3rWFpB-mod:-KTsJStPl9Fpwsq_dFpm:bt2vr1b) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/java/com/vaadin/demo/dashboard/view/transactions/TransactionsView.java#L203)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> Can this change during the loop ? Why was this changed from the previous where it was much more simple
</details>
Fixed.

---

*[src/main/resources/com/vaadin/demo/dashboard/DashboardWidgetSet.gwt.xml, line 38 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-QMn6I7OswMd0EbG:-KTsFy47NKm7sx7gSTBg:b-896fix) ([raw file](https://github.com/vaadin/dashboard-demo/blob/fe59f1072358bff37e539b4e8f10fda7ca00e854/src/main/resources/com/vaadin/demo/dashboard/DashboardWidgetSet.gwt.xml#L38)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> Remove
</details>
Done.

---

*[src/main/webapp/VAADIN/themes/dashboard/dashboard.scss, line 13 at r1](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10#-KTs-_9Q0V6mVj3F1oPd:-KTsFwm95QbkJ0Cu2wb4:b-896fix) ([raw file](https://github.com/vaadin/dashboard-demo/blob/e38a66d624bbc8d359042e6ce1990940d8adf8d4/src/main/webapp/VAADIN/themes/dashboard/dashboard.scss#L13)):*
<details><summary><i>Previously, pleku (Pekka Hyvönen) wrote…</i></summary>
> Can table be now removed ?
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10)*
<!-- Sent from Reviewable.io -->

pleku commented:
Reviewed 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/dashboard-demo/10)*
<!-- Sent from Reviewable.io -->
